src/Product.java

```diff
@@ -0,0 +1,57 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+
+public class Product {
+    private String type;
+    private double price;
+    private boolean discount;
+
+    public Product(String type, double price, boolean discount) {
+        this.type = type;
+        this.price = price;
+        this.discount = discount;
+    }
+
+    public Product(String type, double price) {
+        this.type = type;
+        this.price = price;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public double getPrice() {
+        return price;
+    }
+
+    public boolean hasDiscount() {
+        return discount;
+    }
+
+    //#1
+    public static List<Product> filterExpensiveBooks(List<Product> products) {
```

Comment:
sorry, why expensive? The requirement id type Book and price > 250. Confuse a bit